plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/expression/SparkExpressionBuilder.java

```diff
@@ -0,0 +1,231 @@
+    {
+        return new BooleanLiteral(context.getText());
+    }
+
+    @Override
+    public SparkExpression visitIntegerLiteral(SparkExpressionParser.IntegerLiteralContext context)
+    {
+        return new LongLiteral(context.getText());
+    }
+
+    @Override
+    public Object visitUnicodeStringLiteral(SparkExpressionParser.UnicodeStringLiteralContext context)
+    {
+        return new StringLiteral(decodeUnicodeLiteral(context));
+    }
+
+    private static String decodeUnicodeLiteral(SparkExpressionParser.UnicodeStringLiteralContext context)
+    {
+        String rawContent = unquote(context.getText());
+        StringBuilder unicodeStringBuilder = new StringBuilder();
+        StringBuilder escapedCharacterBuilder = new StringBuilder();
+        int charactersNeeded = 0;
+        UnicodeDecodeState state = UnicodeDecodeState.EMPTY;
+        for (int i = 0; i < rawContent.length(); i++) {
+            char ch = rawContent.charAt(i);
+            switch (state) {
+                case EMPTY:
+                    if (ch == SPARK_ESCAPE_CHARACTER) {
+                        state = UnicodeDecodeState.ESCAPED;
+                    }
+                    else {
+                        unicodeStringBuilder.append(ch);
+                    }
+                    break;
+                case ESCAPED:
+                    if (ch == SPARK_ESCAPE_CHARACTER) {
+                        unicodeStringBuilder.append(SPARK_ESCAPE_CHARACTER);
+                        state = UnicodeDecodeState.EMPTY;
+                    }
+                    else if (ch == 'u') {
+                        state = UnicodeDecodeState.UNICODE_SEQUENCE;
+                        charactersNeeded = 4;
+                    }
+                    else if (ch == 'U') {
+                        state = UnicodeDecodeState.UNICODE_SEQUENCE;
+                        charactersNeeded = 8;
+                    }
+                    else if (isHexDigit(ch)) {
+                        state = UnicodeDecodeState.UNICODE_SEQUENCE;
+                        escapedCharacterBuilder.append(ch);
+                    }
+                    else {
+                        throw new ParsingException("Invalid hexadecimal digit: " + ch);
+                    }
+                    break;
+                case UNICODE_SEQUENCE:
+                    checkState(isHexDigit(ch), "Incomplete escape sequence: " + escapedCharacterBuilder, context);
+                    escapedCharacterBuilder.append(ch);
+                    if (charactersNeeded == escapedCharacterBuilder.length()) {
+                        String currentEscapedCode = escapedCharacterBuilder.toString();
+                        escapedCharacterBuilder.setLength(0);
+                        int codePoint = Integer.parseInt(currentEscapedCode, 16);
+                        checkState(Character.isValidCodePoint(codePoint), "Invalid escaped character: " + currentEscapedCode, context);
```

Comment:
nit: isValidCodePoint can be imported statically

Use `%s` placeholders when formatting a message for a `checkState` call